src/components/tl/ucp/tl_ucp_ep.c

```diff
@@ -0,0 +1,74 @@
+#include "tl_ucp.h"
+#include "tl_ucp_ep.h"
+
+ucc_status_t ucc_tl_ucp_connect_ep(ucc_tl_ucp_context_t *ctx,
+                                   ucc_tl_ucp_team_t *team, char *addr_array,
+                                   size_t max_addrlen, int rank)
+{
+    ucp_address_t  *address = (ucp_address_t *)(addr_array + max_addrlen * rank);
+    ucp_ep_params_t ep_params;
+    ucs_status_t    status;
+    ucp_ep_h       *ep;
+    if (team->context_ep_storage) {
+        //TODO get ep from ctx storage ctx->eps[rank_to_ctx_rank] using mappers
+        ep = NULL;
+    } else {
+        ucc_assert(team && team->eps);
+        ep = &team->eps[rank];
+    }
+    if (*ep) {
+        /* Already connected */
+        return UCC_OK;
+    }
+    ep_params.field_mask = UCP_EP_PARAM_FIELD_REMOTE_ADDRESS;
+    ep_params.address    = address;
+
+    status = ucp_ep_create(ctx->ucp_worker, &ep_params, ep);
```

Comment:
Nitpick: Does create ep connect (exchange information) the endpoints or just create? If only creates should we call it create?

Reply:
it is actually connect